apps/site/assets/ts/stop/components/DepartureList.tsx

```diff
@@ -32,22 +40,21 @@ const DepartureList = ({
   // TODO: handle no predictions or schedules case and predictions only case
   return (
     <>
-      {schedules.length && (
-        <div className="stop-departures departure-list-header">
-          <div className={`departure-card__route ${routeBgClass(route)}`}>
-            <div>
-              {renderSvg("c-svg__icon", routeToModeIcon(route), true)}{" "}
-              {routeName(route)}
-            </div>
-            <a
-              className="open-schedule"
-              href={`../schedules/${route.id}/line?schedule_direction[direction_id]=${directionId}&schedule_direction[variant]=${schedules[0].trip.route_pattern_id}`}
-            >
-              View all schedules
-            </a>
+      <div className="stop-departures departure-list-header">
+        <div className={`departure-card__route ${routeBgClass(route)}`}>
+          <div>
+            {renderSvg("c-svg__icon", routeToModeIcon(route), true)}{" "}
+            {routeName(route)}
           </div>
+          <a
+            className="open-schedule"
+            href={`../schedules/${route.id}/line?schedule_direction[direction_id]=${directionId}&schedule_direction[variant]=${schedules[0]?.trip.route_pattern_id}`}
+          >
+            View all schedules
+          </a>
         </div>
-      )}
+      </div>
+      {schedules.length === 0 && displayNoUpcomingTrips()}
```

Comment:
Yeah it should.  I am curious about the weird limbo this page is currently in.  The page doesn't support showing times without schedules.  So if there are no schedules, but there are predictions, this page is displaying nothing.
In this current form they'd at least get a message about no upcoming trips.  Thoughts?

Reply:
I see what you mean! My preference would be to incorporate predictions at this point to take this component out of limbo since this ticket was originally estimated to include the requirement "It should only display the prediction time if it exists, schedule time otherwise". Whether that is done as part of this PR or a separate one I'll leave up to you! 

It seems to me that there may be a small change that could be made here to resolve both issues. For example, what if `mergeIntoDepartureInfo` was pulled up before line 57? Then we could check the length of `departures` instead and iterate over departureInfos directly. What do you think of that approach?